test/integration/cls.test.js

```diff
@@ -140,6 +140,197 @@ if (current.dialect.supports.transactions) {
+            spyMatcher.push(sinon.match.has('transaction', transaction));
+
+            for (const [hookName, spy] of Object.entries(hooks)) {
+              expect(
+                spy,
+                `hook ${hookName} did not receive the transaction from CLS.`,
+              ).to.have.been.calledWith(...spyMatcher);
+            }
+          });
+        });
+      }
+
+      testHooks({
+        method: 'Model.bulkCreate',
+        hooks: ['beforeBulkCreate', 'beforeCreate', 'afterCreate', 'afterBulkCreate'],
+        optionPos: 1,
+        async execute(User) {
+          await User.bulkCreate([{ name: 'bob' }], { individualHooks: true });
+        },
+      });
+
+      testHooks({
+        method: 'Model.findAll',
+        hooks: ['beforeFind', 'beforeFindAfterExpandIncludeAll', 'beforeFindAfterOptions', 'afterFind'],
+        optionPos: 0,
+        async execute(User) {
+          await User.findAll();
+        },
+      });
+
+      testHooks({
+        method: 'Model.count',
+        hooks: ['beforeCount'],
+        optionPos: 0,
+        async execute(User) {
+          await User.count();
+        },
+      });
+
+      testHooks({
+        method: 'Model.upsert',
+        hooks: ['beforeCount', 'afterUpsert'],
+        optionPos: 1,
+        async execute(User) {
+          await User.upsert({
+            name: 'bob',
+          });
+        },
+      });
+
+      testHooks({
+        method: 'Model.destroy',
+        hooks: ['beforeBulkDestroy', 'afterBulkDestroy'],
+        optionPos: 0,
+        async execute(User) {
+          await User.destroy({ where: { name: 'bob' } });
+        },
+      });
+
+      testHooks({
+        method: 'Model.destroy with individualHooks',
+        hooks: ['beforeDestroy', 'beforeDestroy'],
+        optionPos: 1,
+        async execute(User) {
+          await User.create({ name: 'bob' });
+          await User.destroy({ where: { name: 'bob' }, individualHooks: true });
+        },
+      });
+
+      testHooks({
+        method: 'Model#destroy',
+        hooks: ['beforeDestroy', 'beforeDestroy'],
+        optionPos: 1,
+        async execute(User) {
+          const user = await User.create({ name: 'bob' });
+          await user.destroy();
+        },
+      });
+
+      testHooks({
+        method: 'Model.update',
+        hooks: ['beforeBulkUpdate', 'afterBulkUpdate'],
+        optionPos: 0,
+        async execute(User) {
+          await User.update({ name: 'alice' }, { where: { name: 'bob' } });
+        },
+      });
+
+      testHooks({
+        method: 'Model.update with individualHooks',
+        hooks: ['beforeUpdate', 'afterUpdate'],
+        optionPos: 1,
+        async execute(User) {
+          await User.create({ name: 'bob' });
+          await User.update({ name: 'alice' }, { where: { name: 'bob' }, individualHooks: true });
+        },
+      });
+
+      testHooks({
+        method: 'Model#save (isNewRecord)',
+        hooks: ['beforeCreate', 'beforeUpdate'],
```

Comment:
I made a mistake here
```suggestion
        hooks: ['beforeCreate', 'afterCreate'],
```